### PR TITLE
Add some typehints in core/helpers.py

### PR DESCRIPTION
**Pull request by yivgen**
Related to #8028

Adds typehints in core/helpers.py

### Stakeholders

@scottbarnes

### CODE REVIEW

File: openlibrary/core/helpers.py [hunk 1]
```diff
@@ -8,11 +8,17 @@
...
 
+from typing import TYPE_CHECKING
+
 import babel
 import babel.core
 import babel.dates
 import babel.numbers
 
+from babel.core import Locale
+from typing import Optional
```
Comment (scottbarnes) [inline openlibrary/core/helpers.py:19]:
```suggestion
```

File: openlibrary/core/helpers.py [hunk 2]
```diff
@@ -196,7 +212,7 @@ def sprintf(s, *a, **kw):
         return s
 
 
-def cond(pred, true_value, false_value=""):
+def cond(pred, true_value, false_value):
```
Comment (scottbarnes) [inline openlibrary/core/helpers.py:215]:
```suggestion
def cond(pred: Any, true_value: Any, false_value: Any = "") -> Any:
```
Good call on not putting `list[Author]` here, etc., as that could be misleading, given all the possible valid things one could supply here.

My suggestion, using `Any`, is of questionable value, but it least it lets others know the function has been considered vis-a-vis typing.

The main thing is that we'll want to keep the `""` default value for `false_value`, so we don't make `false_value` a required argument for calling functions that may now rely on it having a default.

File: openlibrary/core/helpers.py [hunk 3]
```diff
@@ -31,6 +37,9 @@
 from infogami.infobase.utils import parse_datetime
 from infogami.utils.view import safeint
 
+if TYPE_CHECKING:
+    from openlibrary.plugins.upstream.models import Author
+
```
Comment (scottbarnes) [inline openlibrary/core/helpers.py:42]:
```suggestion
```
Because `Author` isn't used, this won't be needed.

File: openlibrary/core/helpers.py [hunk 4]
```diff
@@ -8,11 +8,17 @@
 
 import web
 
+from typing import TYPE_CHECKING
+
```
Comment (scottbarnes) [inline openlibrary/core/helpers.py:12]:
```suggestion
```
Because `Author` isn't used, this won't be needed.

**Review by scottbarnes — COMMENTED** [review]
Thanks for this, @yivgen. Just a few bits of feedback.

**Comment by yivgen** [comment]
Thank you for the informative feedback! I'll add all these of these changes

**Review by scottbarnes — APPROVED** [review]
Great work. Thanks for this, @yivgen.

**Pull request merged**